cmd/wait-for-github/pr.go

```diff
@@ -0,0 +1,129 @@
+// wait-for-github
+// Copyright (C) 2022, Grafana Labs
+
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the GNU Affero General Public License as published by the Free
+// Software Foundation, either version 3 of the License, or (at your option) any
+// later version.
+
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+// details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/grafana/wait-for-github/internal/github"
+	"github.com/grafana/wait-for-github/internal/utils"
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+type prConfig struct {
+	owner string
+	repo  string
+	pr    int
+}
+
+var (
+	// https://regex101.com/r/nexaWT/1
+	pullRequestRegexp = regexp.MustCompile(`.*github\.com/(?P<owner>[^/]+)/(?P<repo>[^/]+)/pull/(?P<number>\d+)/?.*`)
+	prConf            prConfig
+)
+
+func parsePRArguments(c *cli.Context) error {
+	var owner, repo, number string
+
+	switch {
+	// If a single argument is provided, it is expected to be a pull request URL
+	case c.NArg() == 1:
+		url := c.Args().Get(0)
+		match := pullRequestRegexp.FindStringSubmatch(url)
+		if match == nil {
+			return fmt.Errorf("invalid pull request URL: %s", url)
+		}
+
+		owner = match[1]
+		repo = match[2]
+		number = match[3]
+	// If three arguments are provided, they are expected to be owner, repo, and PR number
+	case c.NArg() == 3:
+		owner = c.Args().Get(0)
+		repo = c.Args().Get(1)
+		number = c.Args().Get(2)
+	// Any other number of arguments is an error
+	default:
+		// If the number of arguments is not as expected, show the usage message and error out
+		// I think we should be able to do `cli.ShowCommandHelp(c, "pr")` here,
+		// but it doesn't work, says "unknown command pr". So we go through the parent command.
+		lineage := c.Lineage()
+		parent := lineage[1]
+		cli.ShowCommandHelpAndExit(parent, "pr", 1)
+
+		// shouldn't get here, the previous line should exit
+		return nil
+	}
+
+	prConf.owner = owner
+	prConf.repo = repo
+
+	n, err := strconv.Atoi(number)
+	if err != nil {
+		return fmt.Errorf("PR must be a number, got '%s'", c.Args().Get(2))
+	}
+	prConf.pr = n
+
+	log.Debugf("Will wait for PR %s/%s#%d", owner, repo, n)
```

Comment:
```suggestion
	log.Infof("Will wait for PR %s/%s#%d", owner, repo, n)
```
I feel like all of your logs currently there should be infos. Debug should be things that are way more verbose, like actual calls to the API. That's the nice thing about a log-level option. If you want to disable logging, you can set the level to warning or error. 

But a tool that logs nothing by default feels a bit weird to me personally

Reply:
I did this, it makes sense!